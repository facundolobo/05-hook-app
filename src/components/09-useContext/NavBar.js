import React from 'react'


import { Link,NavLink } from "react-router-dom"; //importar para utilizarlos 
//la diferencia entre estos es q navLink es que peude establecer una clase si el URL 
//coincide con el Path
export const NavBar = () => {
    return (  //codigo cipoado de https://getbootstrap.com/docs/5.0/components/navbar/   renombre className y los link
     
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
  
    <Link className="navbar-brand" to="/">useContext</Link>
    
    <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
      <div className="navbar-nav">
        <NavLink exact activeClassName="active" className="nav-link" to="/" >Home</NavLink> {/*activeClassName 
        para que se vea como seleccionado */}
        <NavLink exact activeClassName="active" className="nav-link" to="/login" >Login</NavLink>
        <NavLink exact activeClassName="active" className="nav-link" to="/about" >About</NavLink>
      </div>
    </div>
 
</nav>
    )
}
