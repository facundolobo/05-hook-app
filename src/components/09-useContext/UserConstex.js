import { createContext } from "react";

export const UserContext = createContext(null); //creamos un contenedor de variables