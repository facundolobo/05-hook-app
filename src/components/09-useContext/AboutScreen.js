import React, { useContext } from 'react'
import { UserContext } from './UserConstex'

export const AboutScreen = () => {
    const {user,setUser} = useContext(UserContext) //en el use "useContext" buscamos un "UserContext"
    
    
    const handleClick= ()=>{//borra los datos de login
        setUser({});
        
    }
    return (
        <div>
            <h1>AboutScreen</h1>
            <hr/>
            <pre>
                {JSON.stringify(user,null,3)}
            </pre>
            <button
            className="btn btn-warning"
            onClick={handleClick}
            >
                LogOut
            </button>
            
        </div>
    )
}
