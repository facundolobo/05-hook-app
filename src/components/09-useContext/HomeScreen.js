import React, { useContext } from 'react'
import { UserContext } from './UserConstex'


export const HomeScreen = () => {

    const {user} = useContext(UserContext) //en el use "useContext" buscamos un "UserContext"

    console.log(user);
    return (
        <div>
            
            <h1>HomeScreen</h1>
            <hr/> 

            <pre >
                {JSON.stringify(user,null,3)}
            </pre>
        </div>
    )
}
