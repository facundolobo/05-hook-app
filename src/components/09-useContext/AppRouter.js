import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route
  } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <div>
            <Router>
                <div>
                    <Switch>
                        
                        <Route exact path="/about" component={AboutScreen} /> {/**el exact obligas a la coincidencia exacta*/}
                        <Route exact path="/login" component={LoginScreen} />

                        <Route exact path="/" component={HomeScreen} /> {/**costumbre dejar la pagina 
                         * mas general abajo */}

                        
                    </Switch>
                </div>
            </Router>
        </div>
    )
}
