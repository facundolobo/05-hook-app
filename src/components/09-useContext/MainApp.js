import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserConstex'

export const MainApp = () => {
    
    const [user, setUser] = useState({

    });

    return (
        
            <UserContext.Provider value= {{
                user,
                setUser
            }}> {/**asi pongo a UserContext como componente superior, contextrProvider 
             * me sirve para proveer informacion a lo largo de 
             * todos sus componentes hijos
              */}
                <AppRouter/>
            </UserContext.Provider>
       
    )
}
