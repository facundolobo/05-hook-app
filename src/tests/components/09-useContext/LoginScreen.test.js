import { mount, shallow } from 'enzyme';
import React, { useContext } from 'react'
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserConstex';




 describe('Pruebas en <LoginScreen/>', () => {

    const setUser= jest.fn(); //creamos la funcion
        //wrapper //mount
     const wrapper= mount( 

        <UserContext.Provider value={{ //constexto
            setUser
        }}>
           <LoginScreen/>
        </UserContext.Provider> )
     
     


     test('debe verse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();

     })
     test('debe ejecutar el setUser con el argumento esperado ', () => {
         //simular un click con los argumentos esperados
        wrapper.find('button').prop('onClick')()// con esto lo llamo como funci√≥n
        expect(setUser).toHaveBeenCalledWith({
            id:123,
            name: "Fernando"
        });

     })
     
     
 })
 