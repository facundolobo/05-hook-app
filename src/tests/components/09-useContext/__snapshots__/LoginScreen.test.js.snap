// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> debe verse correctamente  1`] = `<LoginScreen />`;
